File: 2095-minimum-number-of-swaps-to-make-the-string-balanced/2095-minimum-number-of-swaps-to-make-the-string-balanced.cpp
class Solution {
public:
    int minSwaps(string s) {
        int ans =0;

        for(int i=0; i<s.size(); i++){
            if(s[i] == '[')ans+=1;
            else {
                if(ans > 0){
                    ans -= 1;
                }else {
                    ans += 1;
                }
            }
        }
        return (ans + 1)/2;
    }
};

File: 2095-minimum-number-of-swaps-to-make-the-string-balanced/2095-minimum-number-of-swaps-to-make-the-string-balanced.py
class Solution:
    def minSwaps(self, s: str) -> int:
        ans = 0

        for i in range(len(s)):
            if s[i] == '[':
                ans += 1
            else:
                if ans > 0:
                    ans -= 1
                else: ans +=1

        return (ans + 1)//2    